auth/service.cc

```diff
@@ -322,13 +327,16 @@ static void validate_authentication_options_are_supported(
         }
     };
 
-    if (options.password) {
```

Comment:
why change order of checks? while it shouldn't cause problems it's better to avoid such changes